### PR TITLE
qubes-dom0-update could check whether /boot is mounted

### PR DESCRIPTION
**Pull request by alimirjamali**
Fixes: https://github.com/QubesOS/qubes-issues/issues/885

### CODE REVIEW

File: dom0-updates/qubes-dom0-update [hunk 1]
```diff
@@ -234,6 +234,43 @@ if [ -n "$CLEAN" ]; then
...
+    local CHOICE
+    # No reason to check further if mount point is already mounted
+    awk -v PART="${1}" '{if ($2 == PART ) { exit 0 }} ENDFILE{exit -1}' < /proc/mounts
+    [[ ${?} -ne 0 ]] || return
+    # No reason to check further if mount point is not in fstab
+    awk -v PART="${1}" '!/^[ \t]*#/{ if ( $2 == PART ) { exit 0}} ENDFILE {exit -1}' < /etc/fstab
+    [[ ${?} -ne 0 ]] && return
+    # Ask user to manually mount partition if user is using GUI Updater
+    if [ ! -t 1 ]; then
+        echo "Could not decide about unmounted ${1} partition in non-interactive/GUI mode!"
```
Comment (marmarek) [inline dom0-updates/qubes-dom0-update:248]: This doesn't allow overriding the check in non-interactive case. Maybe add some option for that?

File: dom0-updates/qubes-dom0-update [hunk 2]
```diff
@@ -234,6 +234,43 @@ if [ -n "$CLEAN" ]; then
...
+                exit 1
+            fi
+            ;;
+        n|N) echo "Warning! Proceeding forward without mounting ${1}";;
+        a|A) echo Operation aborted!; exit 1;;
+        *) echo Invalid choice. Aborting!; exit 1;;
+    esac
+}
+
+if [ "$CHECK_ONLY" != "1" ]; then
```
Comment (marmarek) [inline dom0-updates/qubes-dom0-update:267]: There are more cases when mounting is not necessary or desired: `REMOTE_ONLY=1`, `DOWNLOADONLY=1`. In fact `CHECK_ONLY` implies `REMOTE_ONLY` already.

**Review by marmarek — CHANGES_REQUESTED** [review]

File: dom0-updates/qubes-dom0-update (hunk 1 above)
Comment (alimirjamali) [reply dom0-updates/qubes-dom0-update:248]: > This doesn't allow overriding the check in non-interactive case. Maybe add some option for that?

Is `--skip-boot` or `--skip-boot-check` good? And the help text:

```
--skip-boot       do no check if /boot & /boot/efi are mounted.
```

**Comment by marmarek** [comment]
--skip-boot-check is okay.

-- 
Best Regards,
Marek Marczykowski-Górecki
Invisible Things Lab

**Review by marmarek — APPROVED** [review]

**Pull request merged**